src/contracts/engine/transactions/sellAndBurnMln.ts

```diff
@@ -0,0 +1,70 @@
+import { isSameToken } from '@melonproject/token-math/token';
+import {
+  QuantityInterface,
+  createQuantity,
+  isEqual,
+  greaterThan,
+} from '@melonproject/token-math/quantity';
+import { Address } from '~/utils/types';
+import {
+  prepareTransaction,
+  sendTransaction,
+  getContract,
+} from '~/utils/solidity';
+import { getToken } from '~/contracts/dependencies/token';
+import { isAddress } from '~/utils/checks';
+import { ensure } from '~/utils/guards';
+import { Contracts } from '~/Contracts';
+
+const guards = async (
```

Comment:
Better to use the transactionFactory. Sorry, I know it is quite new, but it makes this stuff much easier.